src/entity/in_memory.ts

```diff
@@ -88,13 +88,15 @@ export class InMemoryEntity {
 
     /**
      * @summary Clone this entity
-     * @param extraContext {Object}
-     * @returns {*}
      */
-    clone(extraContext: object = {}): InMemoryEntity {
-        const Entity = this.constructor as typeof InMemoryEntity;
+    clone<T extends InMemoryEntity>(extraContext?: object): T {
+        // @ts-ignore
+        const object: T = new this.constructor({
```

Comment:
Why do we need to use TS ignore?

Reply:
For some reason, typescript defines `this.constructor` as an ordinary function, not a constructor of some particular type. Everytime I try to access some properties or build a new object via `new this.constructor` it throws an error.
https://stackoverflow.com/questions/33387318/access-to-static-properties-via-this-constructor-in-typescript

I faced this problem in multiple places, and haven't found a proper solution yet